src/proto/streams/streams.rs

```diff
@@ -748,16 +758,13 @@ where
     }
 
     pub fn has_streams_or_other_references(&self) -> bool {
-        if Arc::strong_count(&self.inner) > 1 {
-            return true;
-        }
 
         if Arc::strong_count(&self.send_buffer) > 1 {
```

Comment:
I'll take a closer look to make sure, but yeah, it's quite possible this is entirely redundant.

Reply:
Update: nevermind, removing the `send_buffer` refcount check seems to have broken several of the tests:

```
failures:
    drop_pending_open
    handshake
    http_2_request_without_scheme_or_authority
    notify_on_send_capacity
    recv_too_big_headers
    request_stream_id_overflows
```

I'll look into that.

Reply:
Ah, I see --- it looks like the issue is that, since `Streams` itself is `Clone`, cloning the `Streams` was updating the `Arc` refcount for `send_buffer` but not the `refs` value on `inner`, so these values _were_ different. I'll modify this so that cloning/dropping `Streams` also updates the inner ref count.